scripts/github-translation-status.mjs

```diff
@@ -0,0 +1,462 @@
+			output.debug(`Payload: ${JSON.stringify(payload)}`);
+			return;
+		}
+	}
+
+	async updatePageIndex (oldPages) {
+		// Initialize a new page index with a stable key order
+		const pages = {
+			[this.sourceLanguage]: {},
+		};
+		this.targetLanguages.forEach(lang => pages[lang] = {});
+
+		// Enumerate all markdown pages with supported languages in pageSourceDir,
+		// retrieve their page data and update them
+		const pagePaths = await glob(`**/*.md`, {
+			cwd: this.pageSourceDir,
+		});
+		const updatedPages = await Promise.all(pagePaths.sort().map(async pagePath => {
+			const pathParts = pagePath.split('/');
+			if (pathParts.length < 2)
+				return;
+			const lang = pathParts[0];
+			const subpath = pathParts.splice(1).join('/');
+
+			// Ignore pages with languages not contained in our configuration
+			if (!pages[lang])
+				return;
+
+			// Attempt to get old data for the current page from our index
+			const oldPageData = { ...(oldPages[lang] && oldPages[lang][subpath]) };
+
+			// Create or update page data for the page
+			return {
+				lang,
+				subpath,
+				pageData: await this.updateSinglePageData({
+					pagePath,
+					oldPageData
+				}),
+			};
+		}));
+		
+		// Write the updated pages to the index
+		updatedPages.forEach(page => {
+			if (!page)
+				return;
+			const { lang, subpath, pageData } = page;
+			pages[lang][subpath] = pageData;
+		});
+
+		return pages;
+	}
+
+	/**
+	 * Processes the markdown page located in the pageSourceDir subpath `pagePath`
+	 * and creates a new page data object based on its frontmatter, git history and
+	 * old page data.
+	 */
+	async updateSinglePageData ({ pagePath, oldPageData }) {
+		const fullFilePath = `${this.pageSourceDir}/${pagePath}`;
+		const latest = (a, b) => (a > b ? a : b);
+		const pageData = {};
+
+		// Retrieve git history for the current page
+		const gitHistory = await this.getGitHistory(fullFilePath);
+
+		// Detect and store i18nReady flag from frontmatter
+		const frontMatterBlock = this.tryGetFrontMatterBlock(fullFilePath);
+		const i18nReady = /^\s*i18nReady:\s*true\s*$/m.test(frontMatterBlock);
+		if (i18nReady) {
+			pageData.i18nReady = true;
+			// If the page was i18nReady before, keep the old i18nReadyDate (if any),
+			// or use the last commit date as a fallback
+			pageData.i18nReadyDate = oldPageData.i18nReady &&
+				oldPageData.i18nReadyDate || gitHistory.lastCommitDate;
+		}
+
+		// Use the most recent dates (which allows us to manually set future dates
+		// if we do not want a translated page to become outdated) and the actual commit messages
+		pageData.lastChange = latest(oldPageData.lastChange, gitHistory.lastCommitDate);
+		pageData.lastCommitMsg = gitHistory.lastCommitMessage;
+		pageData.lastMajorChange = latest(oldPageData.lastMajorChange, gitHistory.lastMajorCommitDate);
+		pageData.lastMajorCommitMsg = gitHistory.lastMajorCommitMessage;
+		
+		return pageData;
+	}
+
+	tryGetFrontMatterBlock (filePath) {
```

Comment:
Not sure if this would simplify your life much, but could you use [`gray-matter`](https://www.npmjs.com/package/gray-matter) here? That’s how Astro parses frontmatter, so it’s already a dependency.